src/main/java/edu/byu/cs/autograder/git/GitHelper.java

```diff
@@ -88,8 +88,13 @@ private boolean shouldVerifyCommits() {
     private void fetchRepo(File intoDirectory) throws GradingException {
         gradingContext.observer().update("Fetching repo...");
 
+        fetchRepoFromUrl(gradingContext.repoUrl(), intoDirectory);
+    }
+
+    //Method must be static due to the lack of GradingContext in cases where this method is required
+    public static void fetchRepoFromUrl(String repoUrl, File intoDirectory) throws GradingException {
```

Comment:
It could be helpful to have an overload of this function that automatically generates the temp directory and returns the directory reference as the result.

It's optional, but it would be helpful now and could be helpful in the future too.

Reply:
Can do